src/core/server/http/router/router.ts

```diff
@@ -136,7 +136,7 @@ export class Router {
       );
     }
 
-    return route.validate ? route.validate(schema) : undefined;
+    return route.validate ? route.validate : undefined;
```

Comment:
we can also add a check that all keys for a given object are `@kbn/config-schema` values, don't know if it's worth adding tho 🤔

Reply:
That check could be useful for JS plugins that are migrating over. If we check that they're passing schema objects earlier in the process we can avoid obscure errors for any endpoints still using Joi.

Reply:
such type of errors can be detected by TS on the compilation step. the only potential source of problems are plugins written in js

Reply:
added